Use a common attribute interface (#717)
open-telemetry/opentelemetry-specification: specification/resource/sdk.md

## Resource creation

> A resource is made by a collection of attributes. Resources MUST use the same
> `Attribute` definition used for [Span Attributes](../trace/api.md#set-attributes).

Reviewer: ```suggestion
A resource is a collection of [Attributes](../trace/api.md#set-attributes).
Resources MUST allow the same value types as Spans.
```

Reviewer 2: s/"A resource is a collection of attributes"/"A resource is represented by a collection of attributes"/

> The SDK MUST support two ways to instantiate new resources. Those are:

Reviewer 2: > Those are:

You can't use semicolon to refer to sections. Instead, say "The SDK MUST support two ways to instantiate new resources: Create and Merge". It also makes the sentence self-contained, you don't need to read through to understand what those two ways are.

Required parameters:

- a collection of [attributes](../trace/api.md#set-attributes).

> Attribute values of `null` are considered to be used to create the resource and

Reviewer: I don't understand this sentence.
Should we also reference the null handling of spans here?

Reviewer 2: +1. What is the motivation for this?

Reviewer 3: I suspect this has to do with the pre-existing definition of Merge? `null` is not a valid attribute value, so this seems unnecessary to state. (OTOH, [I'd like it to be a valid attribute value, but it would still be unnecessary.](https://github.com/open-telemetry/opentelemetry-specification/issues/503))

Author: Since we link the definition of Attributes used by `setAttributes`, I wanted to state that for resources, `null` values should be handled slightly differently.
